source/client/process_impl.cc

```diff
@@ -244,7 +250,17 @@ class NighthawkServerFactoryContext : public Envoy::Server::Configuration::Serve
     PANIC("NighthawkServerFactoryContext::healthCheckFailed not implemented");
   }
 
+  void setClusterManager(Envoy::Upstream::ClusterManager& cluster_manager) {
+    cluster_manager_ = &cluster_manager;
+  }
+
+  void setSslContextManager(Envoy::Ssl::ContextManager& ssl_context_manager) {
+    ssl_context_manager_ = &ssl_context_manager;
+  }
+
 private:
+  Envoy::Ssl::ContextManager* ssl_context_manager_;
```

Comment:
Just to be completely safe, let's initialize these to `nullptr`, e.g.
`Envoy::Ssl::ContextManager* ssl_context_manager_ = nullptr;`

and in the getters, `PANIC` if they're nullptr, and only do `*` if nonnull.